### PR TITLE
[PDMO-76] Enable showing of encounters without visits

### PR DESCRIPTION
**Pull request by jacksono**
## JIRA TICKET NAME:

[PDMO-76: Enable showing of encounters without visits](https://issues.openmrs.org/browse/PDMO-76)

## SUMMARY:
The encounters table should be able to show encounters without visits on the same table with the visit column left blank

### CODE REVIEW

**Review by Farhanisto — COMMENTED** [review]
Looks Fine.

File: app/js/components/visitsAndEncounters.jsx [hunk 1]
```diff
@@ -151,34 +152,52 @@ export class VisitsAndEncounters extends React.Component {
...
+      if(firstEncounter.date < SecondEncounter.date)
+        return 1;
+      return 0;
+    }
+
+  return output.sort(compareEncountersbyDate)
+}
+
+render() {
+  console.log("data", this.getAllEncounters())
```
Comment (Ewanjiru) [inline app/js/components/visitsAndEncounters.jsx]: Remove this console statement

Comment (jacksono) [reply app/js/components/visitsAndEncounters.jsx]: Thanks for noticing that, i have removed it

File: app/js/components/visitsAndEncounters.jsx [hunk 2]
```diff
@@ -151,34 +152,51 @@ export class VisitsAndEncounters extends React.Component {
...
-        id: 'visit',
-        Header: 'Visit',
-        accessor: rowProps => <a>{rowProps.visit}</a>,
-      }
-    ];
-    return(
-      <div>
+    function compareEncountersByDate(firstEncounter, secondEncounter) {
+      if(firstEncounter.date > secondEncounter.date)
+        return -1;
```
Comment (dkayiwa) [inline app/js/components/visitsAndEncounters.jsx]: Looking at the general comparison routines, shouldn't 1 be returned for greater and then -1 for less than?

Comment (jacksono) [reply app/js/components/visitsAndEncounters.jsx]: Doesn't that depend on whether you are comparing for ascending or descending order? In this case I wanted a descending order to be consistent with the required reverse chronology. @dkayiwa

Comment (dkayiwa) [reply app/js/components/visitsAndEncounters.jsx]: The method body should never care about whether it is descending on ascending. That is up to the caller. Not so?
Have you looked at any compareTo method in the Java API to see when 1 and -1 are returned?

Comment (jacksono) [reply app/js/components/visitsAndEncounters.jsx]: Let me check it out

**Comment by jacksono** [comment]
@dkayiwa  I have made the correction

**Pull request merged**